tunnel-server/src/app.ts

```diff
@@ -31,4 +40,37 @@ export const app = ({ isProxyRequest, proxyHandlers, logger }: {
     logger,
   })
     .register(fastifyRequestContext)
+    .get<{Querystring: {env: string, returnPath?: string}}>('/login', {
+      schema: {
+        querystring:{
+          properties: {
+            env: { type: 'string' },
+            returnPath: { type: 'string' }
+          },
+          required: ['env']
+        }
+      }
+    }, async function (req, res) {
+      const {env:envId, returnPath = '/'} = req.query
+      if (!returnPath.startsWith('/')){
+        res.statusCode = 400
+        return {error: 'returnPath must be a relative path' }
```

Comment:
nit: relative path does not start with `/`
```suggestion
        return {error: 'returnPath must be an absolute path' }
```

Reply:
understood - still not relative, error is misleading
```suggestion
        return {error: 'returnPath must not be a full URL' }
```